server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -231,4 +243,24 @@ private String convertToString(ArrayList<String> comments) {
 
     return res.toString();
   }
+
+  /**
+   * Filters out a comment string so that it's readable and doesn't break the frontend.
+   *
+   * @param comment The comment to be filtered.
+   * @return A properly formatted comment
+   */
+  private String filterComment(String comment) {
+    String filteredComment = comment;
+
+    // Handles non-ASCII characters, newlines,
+    // trimmed quotes, embedded escaped quotes.
+    // Order matters.
```

Comment:
This is one of those places where you want to do a thorough breakdown of what each part of this code does. Expect to have more comment than code. Something like:
```
// Strip away the following characters
// \u00a9 : copyright character
// \ufoo: foo character
// ...
filteredComment = filteredComment.replaceAll("(\u00a9|\u00ae|[\u2000-\u3300]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])", "");
// Each comment should be on its own line so they are parsed as one large sentence
filteredComment = String.join("\n", filteredComment.split("\\\\n"));
...
```